routing/router_test.go

```diff
@@ -421,9 +421,10 @@ func TestChannelUpdateValidation(t *testing.T) {
 		},
 	}
 
-	route := &Route{
-		Hops: hops,
-	}
+	route := NewRouteFromHops(
```

Comment:
This was always incorrect, because the route wasn't fully populated. The error was exposed because determination of the error source in router changed.